components/function/signature/index.test.tsx

```diff
@@ -7,45 +7,55 @@ import {
   buildOutputList,
 } from "testing/factory";
 import { Signature } from ".";
-import { act } from "@testing-library/react";
+import { ComponentProps } from "react";
+
+const renderSignature = (props: Partial<ComponentProps<typeof Signature>>) => {
+  return render(
+    <Signature
+      func={props.func || buildAbiDefinedFunction()}
+      collapsed={props.collapsed || false}
+      toggleCollapsed={props.toggleCollapsed || vi.fn()}
+    />
+  );
+};
 
 describe("Signature", () => {
-  it("should expand and collapse a function signature", () => {
-    const func = buildAbiDefinedFunction({ inputs: buildInputList(2) });
-    let collapsed = true;
-    let setCollapsed = () => {
-      collapsed = !collapsed;
-    };
-
-    render(
-      <Signature
-        func={func}
-        collapsed={collapsed}
-        setCollapsed={setCollapsed}
-      />
-    );
-
-    expect(screen.findAllByText(func.inputs[0].name!)).resolves.toHaveLength(0);
-    act(() => {
-      screen.getByTestId(`signature-toggle-collapse-${func.name}`).click();
-    });
-    expect(screen.findAllByText(func.inputs[0].name!)).resolves.toHaveLength(1);
```

Comment:
This test would pass even without the `act`, likely due to the use of the `find*` testing-library matcher without a corresponding `await`/`waitFor`. Tests asserting the proper conditional rendering corresponding with the collapsed state were added elsewhere, since the logic for managing the collapsed/expanded state is handled by an ancestor of this component.

As such, this test was replaced with three that make the following assertions:

- The collapse button is shown when the function is expanded
- The expand button is show when the function is collapsed
- Clicking the toggle button calls the provided `toggleCollapsed` prop function (changed from `setCollapsed`)